src/platform_linux/platform_inline.h

```diff
@@ -469,43 +484,58 @@ platform_aligned_malloc(const platform_heap_id heap_id,
  *
  * Reallocing to size 0 must be equivalent to freeing.
  * Reallocing from NULL must be equivalent to allocing.
+ *
+ * Returns ptr to reallocated memory fragment. In case of shared memory,
+ *  returns the newsize padded-upto cache-line alignment bytes.
  */
+#define platform_realloc(hid, oldsize, ptr, newsize)                           \
+   platform_do_realloc(                                                        \
+      (hid), (oldsize), (ptr), (newsize), __func__, __FILE__, __LINE__)
+
 static inline void *
-platform_realloc(const platform_heap_id heap_id,
-                 const size_t           oldsize,
-                 void                  *ptr, // IN
-                 const size_t           newsize)       // IN
+platform_do_realloc(const platform_heap_id heap_id,
+                    const size_t           oldsize,
+                    void                  *ptr,     // IN
+                    size_t                *newsize, // IN/OUT
```

Comment:
Reallocation now returns the `*newsize`, so clients like writable buffer resize can record the new fragment's size in its `buffer_capacity` field. 

This, then, allows `writable_buffer_deinit()` to correctly supply the newly realloc'ed fragment's size to `free`.